### PR TITLE
Migrate website to Docusaurus

### PR DESCRIPTION
**Pull request by Pessimistress**
Staged at: https://pessimistress.github.io/react-map-gl/

#### Todo
- [ ] Search

### CODE REVIEW

File: docs/upgrade-guide.md [hunk 1]
```diff
@@ -62,7 +62,7 @@ v7 is a complete rewrite of the library. It is redesigned to be fast, lightweigh
 
 - `MapContext` is now an official API. The experimental `_MapContext` export will be removed in a future release.
 - `react-virtualized-auto-sizer` is no longer a dependency.
-- Inertia has been enabled by default on the map controller. To revert to the behavior in previous versions, set the [interaction options](/docs/api-reference/interactive-map.md#interaction-options):
+- Inertia has been enabled by default on the map controller. To revert to the behavior in previous versions, set the [interaction options](https://github.com/visgl/react-map-gl/tree/6.1-release/docs/api-reference/interactive-map.md#interaction-options):
```
Comment (chrisgervang) [inline docs/upgrade-guide.md:65]: Is the idea that older deprecated pages are available as markdown, but no longer needed on the website?

File: website/ocular-docusaurus-plugin/index.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  context,
+  opts = {
+    resolve: {modules: [], alias: {}},
+    debug: false,
+    module: {},
+    plugins: []
+  }
+) {
+  return {
+    name: 'ocular-docusaurus-plugin',
```
Comment (chrisgervang) [inline website/ocular-docusaurus-plugin/index.js:22]: What's the connection to ocular? Is it involved in building?

File: website/src/examples/layers.mdx [hunk 3]
```diff
@@ -0,0 +1,5 @@
+# Dynamic Styling
+
+import App from 'website-examples/layers/src/app';
```
Comment (chrisgervang) [inline website/src/examples/layers.mdx:3]: nit: rename this example folder to `styling` so its connected to the title?

**Review by chrisgervang — APPROVED** [review]

File: docs/upgrade-guide.md (hunk 1 above)
Comment (Pessimistress) [reply docs/upgrade-guide.md:65]: Docusaurus fails to build if you have invalid links. In this case, the links were written according to the directory structure of an earlier release, and the files no longer exist.

File: website/ocular-docusaurus-plugin/index.js (hunk 2 above)
Comment (Pessimistress) [reply website/ocular-docusaurus-plugin/index.js:22]: The plan is to eventually extract it to a standalone module that we can share across multiple repos.

File: website/src/examples/layers.mdx (hunk 3 above)
Comment (Pessimistress) [reply website/src/examples/layers.mdx:3]: We often link to the examples in issues/discussions and renaming will break them.

File: docs/upgrade-guide.md (hunk 1 above)
Comment (chrisgervang) [reply docs/upgrade-guide.md:65]: Ah, ok good to know. Sounds like we could either link to the repo in all what's new and upgrade guides going forward, or let the build tool tell us when it's time to switch it from the nicer website experience to the repo.

**Pull request merged**